### DISCUSSION TITLE
how to avoid multiple state files?

### DISCUSSION THREAD

**Discussion by tenzone** [GoogleCloudPlatform/terraformer#1231] [General]
My intended use case, in aws, is to tag resources related to a project with `"Project":"PROJECT_NAME"`.  I would like to scan AWS account and apply a filter to match on that tag using `--filter="Name=tags.project;Value=PROJECT_NAME"`.  This works in general, but it creates a state file for all the different resource groups, where i would rather them all be in the same state file, as they should be logically grouped anyhow.  Is there a way to fo this?   

EDIT: i think i did not read the docs too well, as it looks like I can use the putput path to achieve this...

**Comment by sunil29feb** [comment]
Hi @tenzone 
You can use --compact parameter to get it.

Please see documentation below-

Resource structure
Terraformer by default separates each resource into a file, which is put into a given service directory.

The default path for resource files is {output}/{provider}/{service}/{resource}.tf and can vary for each provider.

It's possible to adjust the generated structure by:

Using --compact parameter to group resource files within a single service into one resources.tf file
Adjusting the --path-pattern parameter and passing e.g. --path-pattern {output}/{provider}/ to generate resources for all services in one directory
It's possible to combine --compact --path-pattern parameters together.
